.github/workflows/publish_pypi.yml

```
@@ -113,74 +109,154 @@ jobs:
+                cp $lib_dir/$lib $HOME/.local/lib/
+                install_name_tool -id $HOME/.local/lib/$lib $HOME/.local/lib/$lib
+                codesign --force --sign - $HOME/.local/lib/$lib
+              done
+            fi
+
+            export SDKROOT=${SDKROOT:-$(xcrun --show-sdk-path)}
+
+            # Can't download LLVM-OpenMP directly, use conda/mamba and set environment variables
+            brew install miniforge
+            mamba create -n pybamm-dev $OPENMP_URL
+            if [[ $(uname -m) == "x86_64" ]]; then
+              PREFIX="/usr/local/Caskroom/miniforge/base/envs/pybamm-dev"
+            elif [[ $(uname -m) == "arm64" ]]; then
+              PREFIX="/opt/homebrew/Caskroom/miniforge/base/envs/pybamm-dev"
+            fi
+
+            # Copy libomp.dylib from PREFIX to $HOME/.local/lib, needed for wheel repair
+            cp $PREFIX/lib/libomp.dylib $HOME/.local/lib/
+            install_name_tool -id $HOME/.local/lib/libomp.dylib $HOME/.local/lib/libomp.dylib
+            codesign --force --sign - $HOME/.local/lib/libomp.dylib
+
+            export CC=/usr/bin/clang
+            export CXX=/usr/bin/clang++
+            export CPPFLAGS="$CPPFLAGS -Xpreprocessor -fopenmp"
+            export CFLAGS="$CFLAGS -I$PREFIX/include"
+            export CXXFLAGS="$CXXFLAGS -I$PREFIX/include"
+            export LDFLAGS="$LDFLAGS -L$PREFIX/lib -lomp"
+
+            # cibuildwheel not recognising its environment variable, so set manually
+            export CIBUILDWHEEL="1"
+
+            python scripts/install_KLU_Sundials.py
+            python -m cibuildwheel --output-dir wheelhouse
         env:
+          CIBW_ARCHS_MACOS: auto
           CIBW_BEFORE_BUILD: python -m pip install cmake casadi setuptools wheel delocate
-          CIBW_REPAIR_WHEEL_COMMAND: delocate-listdeps {wheel} && delocate-wheel -v -w {dest_dir} {wheel}
-          CIBW_TEST_COMMAND: python -c "import pybamm; pybamm.IDAKLUSolver()"
+          CIBW_REPAIR_WHEEL_COMMAND: |
+            if [[ $(uname -m) == "x86_64" ]]; then
+              delocate-listdeps {wheel} && delocate-wheel -v -w {dest_dir} {wheel}
+            elif [[ $(uname -m) == "arm64" ]]; then
+              # Use higher macOS target for now: https://github.com/casadi/casadi/issues/3698
+              delocate-listdeps {wheel} && delocate-wheel -v -w {dest_dir} {wheel} --require-target-macos-version 11.1
+              for file in {dest_dir}/*.whl; do mv "$file" "${file//macosx_11_1/macosx_11_0}"; done
+            fi
+          CIBW_TEST_COMMAND: |
+            set -e -x
+            python -c "import pybamm; pybamm.IDAKLUSolver()"
 
-      - name: Upload wheels for macOS arm64
+      - name: Upload wheels for macOS (amd64, arm64)
         uses: actions/upload-artifact@v4
         with:
-          name: wheels_macos_arm64
+          name: wheels_macos_${{ matrix.os }}
```

```suggestion
          name: wheels_${{ matrix.os }}
```

`macos` being repeated, shows up twice